plugins/plugin_utils/lookup.py

```diff
@@ -12,109 +12,90 @@
     },
     "datastore": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "datastores": "datastores",
-                "folders": "folders",
-                "names": "names",
-                "types": "types",
-            }
-        }
+        "parent_folders": "folders",
     },
     "vm": {
-        "list": {
-            "query": {
-                "clusters": "clusters",
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "hosts": "hosts",
-                "names": "names",
-                "resource_pools": "resource_pools",
-                "vms": "vms",
-            }
-        }
-    },
-    "network": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "names": "names",
-                "networks": "networks",
-                "types": "types",
-            }
-        }
+        "parent_folders": "folders",
     },
+    "network": {"parent_folders": "folders"},
 }
 
 
+class VcenterApi:
+    def __init__(self, hostname, session):
+        self.hostname = hostname
+        self.session = session
+
+    async def fetch(self, url):
+        async with self.session.get(url) as response:
+            result = await response.json()
+            return result
+
+    def build_url(self, object_type, filters):
+        corrected_filters_for_query = self.correct_filter_names(filters, object_type)
+        if object_type == "resource_pool":
+            object_type = object_type.replace("_", "-")
+
+        return (f"https://{self.hostname}/api/vcenter/{object_type}") + gen_args(
+            corrected_filters_for_query, corrected_filters_for_query.keys()
+        )
+
+    def correct_filter_names(self, filters, object_type):
+        """
+        Objects in vSphere have slightly different filter names. For example, some use 'parent_folders' and some use 'folders'.
+        Its easier to read the code if we do all of the filter corrections at the end using a map.
+        Params:
+            filters: dict, The active filters that should be applied to the REST request
+        """
+        if object_type not in FILTER_MAPPINGS.keys():
+            raise AnsibleLookupError(
+                "object_type must be one of [%s]."
+                % ", ".join(list(FILTER_MAPPINGS.keys()))
+            )
+        corrected_filters = {}
+        for filter_key, filter_value in filters.items():
+            try:
+                corrected_filters[FILTER_MAPPINGS[object_type][filter_key]] = (
+                    filter_value
+                )
+            except KeyError:
+                corrected_filters[filter_key] = filter_value
+
+        return corrected_filters
+
+    async def fetch_object_with_filters(self, object_type, filters):
+        _url = self.build_url(object_type, filters)
+        res = await self.fetch(_url)
```

Comment:
why twice?

Reply:
i think the old implementation of the plugin used both methods. Theres no reason for two different methods now though, ill combine them